src/main/java/io/github/zap/commons/keyvaluegetters/AbstractKeyValueGetter.java

```diff
@@ -0,0 +1,146 @@
+package io.github.zap.commons.keyvaluegetters;
+
+import io.github.zap.commons.utils.NonParameterizedType;
+import io.github.zap.commons.utils.ReflectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Base class for some {@link KeyValueGetter}, contains the default process to locate, deserialize and bind to fields
+ */
+public abstract class AbstractKeyValueGetter implements KeyValueGetter {
+    protected final Builder builder;
+    private final Map<Class<?>, Set<KeyField>> cachedKeyFields = new HashMap<>();
+
+    protected AbstractKeyValueGetter(Builder builder) {
+        this.builder = builder;
+    }
+
+    @Override
+    public <T> T get(Class<T> type) throws GetFailedException {
+        try {
+            T object = type.getConstructor().newInstance();
+            bind(object);
+            return object;
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException | InstantiationException e) {
+            throw new GetFailedException("Unexpected failure: ", e);
+        }
+    }
+
+    @Override
+    public void bind(Object o) throws GetFailedException {
```

Comment:
i have to say, i don't really like that the usage of var which is inconsistent with the rest of our codebase (that doesn't use var)

Reply:
lol I started to use it everywhere. Fixing them kinda takes some time, should we change all to var or to explicit type?

Reply:
it is easier IMO to change a relatively small # of usages of var than our entire project of 40000+ lines across repos

Reply:
well, we are rewriting a lot of stuff later, so this is an opportunity to use the new language feature?

Reply:
we are rewriting a lot of stuff, yes, but there's also a lot of stuff that isn't going to be totally rewritten (my AI lib, for example, which is like 3-4k lines on its own i think); so there's lots of stuff that we would need to go back to and add var for consistency

also IMO var hurts readability a lot; for example in this line

`var rawValue = getResult.getRight().result().get();`

i have no idea what rawValue actually is (i would need to go back and look at how getResult is declared, which isn't the hardest thing in the world, but it adds an unnecessary layer of obfuscation)

of course i believe intelliJ would show you what the type actually is, but when i'm reviewing a PR i typically just read the source in Github. also i would say that relying on IDE features for readability is inherently not a good thing

the only real argument for var is using it in the odd instance where you have an extremely long or complex generic type, but those are rare and i've never really seen an instance where i thought var added readability rather than taking it away

Reply:
ok i gonna revert it to explicit type